packages/server/src/auth/auth.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import * as argon2 from 'argon2';
+import { PrismaService } from '../prisma/prisma.service';
+import { UsersService } from '../users/users.service';
+import { UserSignup } from './types/auth.types';
+
+
+@Injectable()
+export class AuthService {
+  constructor(private usersService: UsersService, private prisma: PrismaService, private jwtService: JwtService) {}
+
+  async validateUsername(username: string, password: string): Promise<any> {
+    const user = await this.usersService.findOneUsername(username);
+
+    if(user && await argon2.verify(user.password, password)) {
+      const { password, username, ...rest } = user;
+      return rest;
+    }
+
+    return null;
+  }
+
+  async validateEmail(email: string, password: string): Promise<any> {
+    const user = await this.usersService.findOneEmail(email);
+
+    if(user && await argon2.verify(user.password, password)) {
+      const { password, email, ...rest } = user;
+      return rest;
+    }
+
+    return null;
+  }
+
+  loginUsername(project_id: string, username: string, password: string) {
+    const payload = { username: username, sub: project_id };
+
+    return { access_token: this.jwtService.sign(payload, { expiresIn: process.env.JWT_EXPIRATION }) };
```

Comment:
@gregfrasco Can we confirm the format of the token?

Reply:
I think for now username and project_id should suffice and we can increase complexity as needed later.